### PR TITLE
APEXMALHAR-2190 Use reusable buffer for serialization in spillable data structures

### PR DESCRIPTION
**Pull request by davidyan74**
@brightchen @tweise @siyuanh  please review

### CODE REVIEW

File: library/src/main/java/org/apache/apex/malhar/lib/utils/serde/AffixKeyValueSerdeManager.java [hunk 1]
```diff
@@ -69,6 +71,9 @@ public Slice serializeDataKey(K key, boolean write)
 
   public V deserializeValue(byte[] buffer, MutableInt offset, int length)
   {
-    return valueSerde.deserialize(buffer, offset, length);
+    Input input = new Input(buffer, offset.intValue(), length);
```
Comment (tweise) [inline library/src/main/java/org/apache/apex/malhar/lib/utils/serde/AffixKeyValueSerdeManager.java]: Do we have to worry about constructing too many objects?

**Review by tweise — COMMENTED** [review]
Looks much cleaner overall!

File: library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableMapImpl.java [hunk 2]
```diff
@@ -138,8 +138,8 @@ public V get(Object o)
       return null;
     }
 
-    tempOffset.setValue(valSlice.offset);
-    return keyValueSerdeManager.deserializeValue(valSlice.buffer, tempOffset, valSlice.length);
+    tmpInput.setBuffer(valSlice.buffer, valSlice.offset, valSlice.length);
+    return keyValueSerdeManager.deserializeValue(tmpInput);
```
Comment (davidyan74) [inline library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableMapImpl.java:142]: @tweise Your concern of creating new objects here has been addressed

**Comment by davidyan74** [comment]
@tweise @brightchen @siyuanh This PR is ready to be merged. Please merge if you see no issues.

**Pull request merged**